### PR TITLE
Upload csv for generating powershell assembly reference docs

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Description
Create a csv file for generating powershell assembly reference docs
- use AzPreview.psd1 as feed file
- update the csv file after AzPreview.psd1generated

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by isra-fel** [comment]
Let me move the pull request to draft state. Please feel free to mark it "Ready for review" when you feel comfortable. Thanks

File: tools/Docs/GenerateDotNetCsv.ps1 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+$feedDir = (Get-Item $FeedPsd1FullPath).Directory
+$feedName = (Get-Item $FeedPsd1FullPath).Name
+Import-LocalizedData -BindingVariable ModuleMetadata -BaseDirectory $feedDir -FileName $feedName
+$modules = $ModuleMetadata.RequiredModules
+
+$dotnetCsv = New-Item -Path "$PSScriptRoot\" -Name "az-ps-latest.csv" -ItemType "file" -Force
+$dotnetCsvContent = ""
+for ($index = 0; $index -lt $modules.Count; $index++){
+    $moduleName = $modules[$index].ModuleName
+    $moduleVersion = if([string]::IsNullOrEmpty($modules[$index].RequiredVersion)){ $modules[$index].ModuleVersion } else{ $modules[$index].RequiredVersion }
```
Comment (isra-fel) [inline tools/Docs/GenerateDotNetCsv.ps1:15]: I believe this script is always run in Powershell 7.0+ , so we can use [Ternary operator ?  : ](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_operators?view=powershell-7#ternary-operator--if-true--if-false) to simplify the script
```suggestion
    $moduleVersion = [string]::IsNullOrEmpty($modules[$index].RequiredVersion)) ? $modules[$index].ModuleVersion : $modules[$index].RequiredVersion
```

File: tools/Docs/az-ps-latest.csv [hunk 2]
```diff
@@ -0,0 +1,123 @@
+pac0,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.Accounts,2.5.1
```
Comment (isra-fel) [inline tools/Docs/az-ps-latest.csv:1]: So this CSV does not have a header row, does it?

File: tools/Docs/az-ps-latest.csv [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+pac113,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.StreamAnalytics,2.0.0
+pac114,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.Subscription,0.8.0
+pac115,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.Support,1.0.0
+pac116,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.Synapse,0.13.0
+pac117,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.TimeSeriesInsights,0.2.0
+pac118,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.TrafficManager,1.0.4
+pac119,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.VMware,0.2.0
+pac120,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.Websites,2.8.0
+pac121,[ps=true;customSource=https://www.powershellgallery.com/api/v2/],Az.WindowsIotServices,0.1.0
+
```
Comment (isra-fel) [inline tools/Docs/az-ps-latest.csv:123]: Just to be careful, is it safe to have an empty line at the end?

**Review by isra-fel — CHANGES_REQUESTED** [review]
I have a few questions:

File: tools/Docs/GenerateDotNetCsv.ps1 (hunk 1 above)
Comment (BethanyZhou) [reply tools/Docs/GenerateDotNetCsv.ps1:15]: Just tried in Pwsh7.1.3, look like it has not supported ternary operator. For compatibility, I prefer to keep the old style.
image

File: tools/Docs/az-ps-latest.csv (hunk 2 above)
Comment (BethanyZhou) [reply tools/Docs/az-ps-latest.csv:1]: Yes, no header row

File: tools/Docs/az-ps-latest.csv (hunk 3 above)
Comment (BethanyZhou) [reply tools/Docs/az-ps-latest.csv:123]: An empty line doesn't matter. This csv has been verified by docs team and was used to generated docs of az-ps-6, which is published now.

File: tools/Docs/GenerateDotNetCsv.ps1 (hunk 1 above)
Comment (BethanyZhou) [reply tools/Docs/GenerateDotNetCsv.ps1:15]: Daily build pipeline still uses 6.2.4 to bump version. This script won't work in this case
image

**Review by isra-fel — APPROVED** [review]

**Pull request merged**